editor.planx.uk/src/pages/FlowEditor/lib/store/user.ts

```diff
@@ -0,0 +1,57 @@
+import { User, UserTeams } from "@opensystemslab/planx-core/types";
+import { Team } from "types";
+import type { StateCreator } from "zustand";
+
+export interface UserStore {
+  id: number;
+  firstName: string;
+  lastName: string;
+  email: string;
+  isPlatformAdmin: boolean;
+  teams: UserTeams[];
+
+  setUser: (user: User) => void;
+  getUser: () => User;
+  canUserEditTeam: (teamSlug: Team["slug"]) => boolean;
+}
+
+export const userStore: StateCreator<UserStore, [], [], UserStore> = (
+  set,
+  get,
+) => ({
+  id: 0,
+  firstName: "",
+  lastName: "",
+  email: "",
+  isPlatformAdmin: false,
+  teams: [],
+
+  setUser: (user: User) =>
+    set({
+      id: user.id,
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      isPlatformAdmin: user.isPlatformAdmin,
+      teams: user.teams,
+    }),
+
+  getUser: () => ({
```

Comment:
I'm not 100% here (worth a try?) but I think we could simplify `setUser()` and `getUser()` here as we don't have any other stores in our `StateCreator` to just - 

```ts
setUser: (user) => set(user),
getUser: () => get(),
```

Reply:
so this pattern `setUser()` & `getUser()` matches exactly how the `TeamStore` is currently setup! I'd vote perhaps one to revisit in a followup PR, as this PR points out a few limitations of how the `TeamStore` is initiated and where it's available too?

Reply:
This works for me!